File: source/aoImportWizard/Views/_blankView.vb
﻿
Imports Contensive.BaseClasses
Imports Contensive.ImportWizard.Models

Namespace Contensive.ImportWizard.Controllers
    Public Class _blankView
        '
        ''' <summary>
        ''' return the next view. 0 goes to the first form (start over)
        ''' </summary>
        ''' <param name="app"></param>
        ''' <returns></returns>
        Public Shared Function processView(app As ApplicationModel, srcViewId As Integer) As Integer
            Try
                Dim cp As CPBaseClass = app.cp
                Dim Button As String = app.cp.Doc.GetText(RequestNameButton)
                If String.IsNullOrEmpty(Button) Then Return srcViewId
                If Button = ButtonCancel Then
                    '
                    ' Cancel
                    ImportConfigModel.clear(app)
                    Return viewIdReturnBlank
                End If
                If Button = ButtonRestart Then
                    '
                    ' Restart
                    ImportConfigModel.clear(app)
                    Return viewIdSelectSource
                End If
                '
                Select Case Button
                    Case ButtonBack
                        '
                        ' -- back to select source
                        Return viewIdSelectSource
                    Case ButtonContinue
                        '
                        ' -- upload the file and continue
                        Dim Filename As String = app.cp.Doc.GetText(RequestNameImportUpload)
                        If String.IsNullOrEmpty(Filename) Then Return viewIdSelectSource
                        '
                        Dim importConfig As ImportConfigModel = ImportConfigModel.create(app)
                        app.cp.TempFiles.SaveUpload(RequestNameImportUpload, "upload", importConfig.privateUploadPathFilename)
                        importConfig.save(app)
                        '
                        Return viewIdSelectContent
                End Select
                Return viewIdUpload
            Catch ex As Exception
                app.cp.Site.ErrorReport(ex)
                Throw
            End Try
        End Function
        '
        ''' <summary>
        ''' return the html for this view
        ''' </summary>
        ''' <param name="app"></param>
        ''' <returns></returns>
        Public Shared Function getView(app As ApplicationModel) As String
            Try
                Dim cp As CPBaseClass = app.cp
                Dim headerCaption As String = "Import Wizard"
                Dim description As String = ""
                Dim content As String = ""
                Dim importConfig As ImportConfigModel = ImportConfigModel.create(app)







                Return HtmlController.createLayout(cp, headerCaption, description, content, True, True, True, True)

            Catch ex As Exception
                app.cp.Site.ErrorReport(ex)
                Throw
            End Try
        End Function
    End Class
End Namespace

File: source/aoImportWizard/Views/_blankView.cs
using System;
using Contensive.ImportWizard.Models;

namespace Contensive.ImportWizard.Controllers {
    public class _blankView {
        // 
        /// <summary>
        /// return the next view. 0 goes to the first form (start over)
        /// </summary>
        /// <param name="app"></param>
        /// <returns></returns>
        public static int processView(ApplicationModel app, int srcViewId) {
            try {
                var cp = app.cp;
                string Button = app.cp.Doc.GetText(constants.RequestNameButton);
                if (string.IsNullOrEmpty(Button))
                    return srcViewId;
                if ((Button ?? "") == constants.ButtonCancel) {
                    // 
                    // Cancel
                    ImportConfigModel.clear(app);
                    return constants.viewIdReturnBlank;
                }
                if ((Button ?? "") == constants.ButtonRestart) {
                    // 
                    // Restart
                    ImportConfigModel.clear(app);
                    return constants.viewIdSelectSource;
                }
                // 
                switch (Button ?? "") {
                    case constants.ButtonBack: {
                            // 
                            // -- back to select source
                            return constants.viewIdSelectSource;
                        }
                    case constants.ButtonContinue: {
                            // 
                            // -- upload the file and continue
                            string Filename = app.cp.Doc.GetText(constants.RequestNameImportUpload);
                            if (string.IsNullOrEmpty(Filename))
                                return constants.viewIdSelectSource;
                            // 
                            var importConfig = ImportConfigModel.create(app);
                            string argreturnFilename = importConfig.privateUploadPathFilename;
                            app.cp.TempFiles.SaveUpload(constants.RequestNameImportUpload, "upload", ref argreturnFilename);
                            importConfig.privateUploadPathFilename = argreturnFilename;
                            importConfig.save(app);
                            // 
                            return constants.viewIdSelectContent;
                        }
                }
                return constants.viewIdUpload;
            } catch (Exception ex) {
                app.cp.Site.ErrorReport(ex);
                throw;
            }
        }
        // 
        /// <summary>
        /// return the html for this view
        /// </summary>
        /// <param name="app"></param>
        /// <returns></returns>
        public static string getView(ApplicationModel app) {
            try {
                var cp = app.cp;
                string headerCaption = "Import Wizard";
                string description = "";
                string content = "";
                var importConfig = ImportConfigModel.create(app);







                return HtmlController.createLayout(cp, headerCaption, description, content, true, true, true, true);

            } catch (Exception ex) {
                app.cp.Site.ErrorReport(ex);
                throw;
            }
        }
    }
}